### DISCUSSION TITLE
Filling in `DESC` column

### DISCUSSION THREAD

**Discussion by CGenie** [dtcenter/METplus#2293] [Configuration] [answered]
Hello,

I'm trying to make a pointstat analysis using metplus. I have ascii files with station names, I use `ascii2nc` to convert them to .nc files. Then I run point stat and get `_sl1l2.txt` files. Currently, the `DESC` column has `NA` in them. Is there a way to pass the station description into the `DESC` column? I see the `POINT_STAT_DESC` variable in config files but I'm not sure how to fill that to use the observation description. If I specify that as `POINT_STAT_DESC = {obs_sid}` I get literal `{obs_sid}`.

**Comment by JohnHalleyGotway** [comment]
Hello @CGenie, I see that you have a question about setting the `DESC` column in the SL1L2 lines generated by the Point-Stat tool. In general, Point-Stat aggregates together multiple matched pairs when computing partial sums (e.g. [SL1L2 line type](https://met.readthedocs.io/en/latest/Users_Guide/point-stat.html#id19)) or statistics (e.g. [CNT line type](https://met.readthedocs.io/en/latest/Users_Guide/point-stat.html#id10)). That is to say that *usually* the `TOTAL` column in those lines has a value > 1.

When aggregating multiple lines together, the logic for picking a single station name to populate the `DESC` column doesn't make sense. Generally, you'd have many station names, one for each station being aggregated. However, based on your question, I'm guessing that you've configured Point-Stat in such a way that you're writing SL1L2 lines consisting of a single matched pair.

The short answer to your question is no, there's no mechanism directly within Point-Stat to detect the case when only a single matched pair is used to create an SL1L2 line and then use it's station name to populate the `DESC` column. However, there's a few other details/options that may be useful to you.

1. The most straight-forward info about individual matched pairs is contained in the [MPR line type](https://met.readthedocs.io/en/latest/Users_Guide/point-stat.html#id24). In there, the `OBS_SID` column already contains the station name. However, I presume there's some good reason that you want SL1L2 lines instead of MPR lines.

2. You *could* run the Stat-Analysis tool on the MPR output from Point-Stat, converting it to SL1L2 *and* populating the `DESC` in exactly the way you want. The `-set_hdr` job command option can be used to override output columns either with a constant string or referencing one of the values from the input columns.

Here's an example:
```
stat_analysis -lookin mpr.txt \
-job aggregate_stat -line_type MPR -out_line_type SL1L2 \
-by OBS_SID -set_hdr DESC OBS_SID -out_stat sl1l2.txt
```
And here's the input [mpr.txt](https://github.com/dtcenter/METplus/files/12281713/mpr.txt) and output 
[sl1l2.txt](https://github.com/dtcenter/METplus/files/12281716/sl1l2.txt) files.

So this does achieve the desired result but requires an extra post-processing step through Stat-Analysis. In this command, the `-set_hdr DESC OBS_SID` option is doing the work... reading the value from the input `OBS_SID` column of the MPR line type and writing it the output `DESC` column of the SL1L2 line type.

3. The Point-Stat [mask](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#mask) configuration option defines how masking regions are defined. The `sid` (or station id) option enables you to list individual station names to be used. If you set `sid` to a list station names, then each will be processed as its own *masking region*. For example,
```
mask = {
...
   sid = [ "71069", "71108", "71114", "71118", "71123", "71207", "71450", "71455", "71457", "71473" ];
...
}
```
Then you'd get SL1L2 output separately for each one of these masks, where each mask consists of a single location. And this station id name will appear in the `VX_MASK` output column. If you have a relatively small set of stations and already know what they are by name, this could be a useful option. If the number of stations is large, this would be less useful and tedious to setup in a configuration file.

I can imagine how Point-Stat could be enhanced to do exactly what you're asking. However, I would still describe this as being an edge case. The intention of Point-Stat is to aggregate data across multiple points to compute partial sums and/or statistics. Information about individual matched pairs is available in the MPR line type. Perhaps we should step back and see why you're using SL1L2 lines rather than MPR lines and see if we can make it possible or easier to use the MPR lines instead?

**Comment by CGenie** [comment]
Hello,

Thank you for such a detailed answer!

Actually, I'm a beginner with metpluls so I'm not sure I know what I'm doing exactly :)

Anyways, about MPR, for some reason I get `NA` values for all the `CLIMO_` columns. Can this be debugged in some way? It's strange because my sl1l2 values are OK.

**Reply by JohnHalleyGotway** [reply, marked as answer]
@CGenie, there are many choices of climatology that folks might use and, by default, none are used. Just like forecast and observation data, users can configure what climatological data is used. Generally speaking, climo data is specified in METplus using [climatological mean fields](https://metplus.readthedocs.io/en/latest/Users_Guide/glossary.html#term-TOOL-NAME-_CLIMO_MEAN_FIELD) and, optionally, [climatological standard deviation fields](https://metplus.readthedocs.io/en/latest/Users_Guide/glossary.html#term-TOOL-NAME-_CLIMO_STDEV_FIELD).

Here's a [METplus Use Case](https://metplus.readthedocs.io/en/latest/generated/model_applications/medium_range/GridStat_fcstGFS_obsGFS_climoNCEP_MultiField.html#sphx-glr-generated-model-applications-medium-range-gridstat-fcstgfs-obsgfs-climoncep-multifield-py) that demonstrates specifying the climo files:
```
GRID_STAT_CLIMO_MEAN_INPUT_DIR = {INPUT_BASE}/model_applications/medium_range/grid_to_grid/nwprod/fix
GRID_STAT_CLIMO_MEAN_INPUT_TEMPLATE = cmean_1d.1959{valid?fmt=%m%d}
```
Looks like we're actually pretty limited on METplus examples of specifying both climo mean and standard deviation. FYI, the climo files we use for testing (and that are also used by NOAA/EMC for verifying their model output) can be found here:
   https://dtcenter.ucar.edu/dfiles/code/METplus/MET/MET_unit_test/unit_test/climatology_data/

In particular, NCEP is mostly using this [Daily 1.0 degree](https://dtcenter.ucar.edu/dfiles/code/METplus/MET/MET_unit_test/unit_test/climatology_data/NCEP_1.0deg/) global data. The cmean files contain climatological means and the cstdv files contain the standard deviations. The files are in GRIB1 format and contain data for several variables (HGT, PRMSL, PWAT, TMAX, TMIN, TMP, UGRD, VGRD) and vertical levels (10 m above gnd, 1000 mb, 2 m above gnd, 250 mb, 500 mb, 700 mb, 850 mb, MSL, atmos col).

**Reply by JohnHalleyGotway** [reply]
@CGenie wanted to followup on this discussion from a few months ago. Can we mark this as being answered or is more discussion needed?

**Reply by CGenie** [reply]
Yes, you can close this. Thank you.
